src/workerd/api/streams/internal-test.c++

```diff
@@ -230,7 +230,8 @@ KJ_TEST("WritableStreamInternalController queue size assertion") {
     // allowed to be queued.
 
     jsg::Ref<ReadableStream> source = ReadableStream::constructor(env.js, kj::none, kj::none);
-    jsg::Ref<WritableStream> sink = jsg::alloc<WritableStream>(env.context, kj::heap<MySink>());
```

Comment:
Would be helpful to include a test here with a mocked implementation of the observer to verify that the new instrumentation points work as expected would be helpful.